src/compute.rs

```diff
@@ -0,0 +1,215 @@
+                                    betweenness_count[*b as usize] += 1;
+                                }
+                            }
+                        }
+                        queued_for_next_round.push(newpath);
+                    }
+                }
+            }
+
+            // prep for next round, start fresh queue list
+            path_list.clear();
+            // load up the queue list, marked as visited
+            for path in queued_for_next_round {
+                let end_index = path[path.len() - 1];
+                path_list.push(path);
+                visited[end_index as usize] = true;
+            }
+
+            // now we do bookkeeping for any found
+            // shortest paths.
+            for f in found_for_this_pathlen {
+                num_paths[f as usize] += 1;
+                num_paths[index] += 1;
+                total_path_length[f as usize] += pathlen;
+                total_path_length[index] += pathlen;
+                search_list.retain(|&x| x != f);
+                search_state[f as usize] = true;
+                if f == j {
+                    done = true;
+                }
+            }
+
+            // If no connection exists, stop searching for it.
+            if !touched {
+                search_list.retain(|&x| x != j);
+                search_state[j as usize] = true;
+                done = true
+            }
+
+            pathlen += 1;
+        }
+    }
+}
+
+/// this function is the thread task
+/// grabs next unprocessed node
+/// if no more nodes, exits
+/// returning betweenness, total path lengths, and num paths
+fn betweenness_task(
+    acounter: Arc<Mutex<usize>>,
+    aindices: Arc<Vec<Vec<GraphIndex>>>,
+) -> (Vec<u32>, Vec<u32>, Vec<u32>) {
+    let start = Instant::now();
+    let indices = &aindices;
+    let num_nodes = indices.len();
+
+    // each worker thread keeps its own cache of data
+    // these are returned when the thread finishes
+    // and then summed by the caller
+    let mut betweenness_count: Vec<u32> = vec![0; num_nodes];
+    let mut total_path_length: Vec<u32> = vec![0; num_nodes];
+    let mut num_paths: Vec<u32> = vec![0; num_nodes];
+
+    let mut finished = false;
+    while !finished {
+        let mut counter = acounter.lock().unwrap();
+        let index: usize = *counter;
+        *counter += 1;
+        drop(counter);
+        if index < num_nodes - 1 {
+            if index % 100 == 0 {
+                println!("node: {}, time: {:?}", index, start.elapsed());
+            }
+            betweenness_for_node(
+                index,
+                indices,
+                &mut betweenness_count,
+                &mut total_path_length,
+                &mut num_paths,
+            );
+        } else {
+            finished = true;
+        }
+    }
+    (betweenness_count, total_path_length, num_paths)
+}
+
+pub fn compute_betweenness(
```

Comment:
Should this be public? 
Docs here would also be nice.

Reply:
Regarding public:  yes, it must be public.  It's called from `graph.rs` 
However, the module `compute` does **not** have to be public, so I've changed that.
```
mod compute;
pub mod edge;
pub mod graph;
```

Reply:
docs done.